packages/react-components/react-tree/src/components/TreeItem/TreeItem.tsx

```diff
@@ -7,7 +7,13 @@ import type { ForwardRefComponent } from '@fluentui/react-utilities';
 import { useTreeItemContextValues_unstable } from './useTreeItemContextValues';
 
 /**
- * TreeItem component - TODO: add more docs
+ * The `TreeItem` component represents a single item in a tree.
```

Comment:
We need to emphasise here that this component will expect a certain order of the children to work properly (first child is them node itself, second child is a nested subtree)

Reply:
probably just mention that it's the same order as document traversal for the treewalker https://developer.mozilla.org/en-US/docs/Web/API/TreeWalker, it might be more concrete

Reply:
Thanks both.. I changed it to this, which should provide enough info + reference:
```
/**
 * The `TreeItem` component represents a single item in a tree.
 * It expects a certain order of children to work properly: the first child should be the node itself,
 * and the second child should be a nested subtree in the form of another Tree component or a standalone TreeItem.
 * This order follows the same order as document traversal for the TreeWalker API in JavaScript:
 * https://developer.mozilla.org/en-US/docs/Web/API/TreeWalker.
 * The content and layout of a TreeItem can be defined using the TreeItemLayout or TreeItemPersonaLayout component,
 * which should be used as a direct child of TreeItem.
 *
 * When a TreeItem has nsted child subtree, an expand/collapse control is displayed,
 * allowing the user to show or hide the children.
 */
```